felix/dataplane/linux/vxlan_mgr.go

```diff
@@ -454,6 +532,11 @@ func (m *vxlanManager) KeepVXLANDeviceInSync(mtu int, xsumBroken bool, wait time
 			if m.getNoEncapRouteTable() == nil {
 				noEncapRouteTable := m.noEncapRTConstruct([]string{"^" + parent.Attrs().Name + "$"}, 4, false, m.dpConfig.NetlinkTimeout, m.dpConfig.DeviceRouteSourceAddress,
```

Comment:
is calling `noEncapRTConstruct()` costly? I think we could use m.ipVersion and a condition on `m.dpConfig.DeviceRouteSourceAddress` vs. `m.dpConfig.DeviceRouteSourceAddressIPv6` and have just one call (the other args seem to be identical, right?)

Reply:
It's not too expensive, but you're right, we should call it less since there is a little bit of looping through regexes and such that we can avoid.